src/rust-crypto/CrossSigningIdentity.ts

```diff
@@ -108,13 +116,20 @@ export class CrossSigningIdentity {
      *   * Upload the private keys to SSSS, if it is set up
      */
     private async resetCrossSigning(authUploadDeviceSigningKeys?: UIAuthCallback<void>): Promise<void> {
+        // XXX: We must find a way to make this atomic, currently if the user do not remember is account password
+        // or 4S passphrase/key the process will fail in a bad state, with keys rotated but not uploaded or saved in 4S.
         const outgoingRequests: Array<OutgoingRequest> = await this.olmMachine.bootstrapCrossSigning(true);
 
+        // If 4s is configured we need to udpate it.
+        const hasKey = await this.secretStorage.hasKey();
+        if (hasKey) {
```

Comment:
should this be called `secretStorageIsReady` or something, rather than `hasKey`? "key" is a bit unclear: there are lots of keys involved here.

Or just inline it:

```suggestion
        if (await this.secretStorage.hasKey()) {
```